tests/ecmascript/test-bi-object-assign.js

```diff
@@ -0,0 +1,147 @@
+/*
+ *  Object.assign() (E7 19.1.2.1)
+ *  http://www.ecma-international.org/ecma-262/7.0/#sec-object.assign
+ */
+
+/*===
+basic tests
+true
+maggie Kittycow Machel Fatty Whale 812 1208 9001
+0,1,2,pig,cow,ape,whale
+maggie 1208 9001 Vegeta
+cow,pig,ape,saiyan
+undefined
+this belongs here
+undefined
+prop1
+get
+set 812
+===*/
+
+function basicTest() {
+    var obj, src, dest;
+
+    // If no coercion is needed, target object is returned (i.e. not a copy).
+    obj = {};
+    dest = Object.assign(obj, { test: "test" });
+    print(dest === obj);
+
+    // Properties are assigned in [[OwnPropertyKeys]] order, moving from left
+    // to right in the argument list:
+    //     1. Array indices (sequential, ascending)
+    //     2. String keys (creation order)
+    //     3. Symbol keys (creation order)
+    // This behavior is easy to test for by assigning from multiple sources to
+    // an empty object and then calling Reflect.ownKeys() on the result.
+    obj = Object.assign({},
+        { pig: "maggie", cow: "Kittycow", ape: "Machel" },
+        [ 812, 1208, 9001 ],
+        { whale: "Fatty Whale" });
+    print(obj.pig, obj.cow, obj.ape, obj.whale, obj[0], obj[1], obj[2]);
+    print(Reflect.ownKeys(obj));
+
+    // Existing properties in the target object can be overwritten.
+    obj = Object.assign({ cow: 1208, pig: 812, ape: 9001 },
+        { pig: "maggie", saiyan: "Vegeta" });
+    print(obj.pig, obj.cow, obj.ape, obj.saiyan);
+    print(Reflect.ownKeys(obj));
+
+    // Only enumerable own properties of the source(s) are used.
+    src = Object.create({ inherited: "something stupid happened" }, {
+        prop1: { enumerable: true,  value: "this belongs here" },
+        prop2: { enumerable: false, value: "something stupid happened" }
+    });
+    obj = Object.assign({}, src);
+    print(obj.inherited);
+    print(obj.prop1);
+    print(obj.prop2);
+    print(Reflect.ownKeys(obj));
+
+    // Accessors in both target and source object(s) are called.
+    obj = Object.create(Object.prototype, {
+        prop: { set: function(value) { print("set", value); }}
+    });
+    src = Object.create(Object.prototype, {
+        prop: { enumerable: true, get: function() { print("get"); return 812; }}
+    });
+    Object.assign(obj, src);
+}
+
+try {
+    print("basic tests");
+    basicTest();
+} catch (e) {
+    print(e.stack || e);
+}
+
+/*===
+coercion tests
+true
+something
+true
+TypeError
+TypeError
+===*/
+
+function coercionTest() {
+    // The first argument to Object.assign() is ToObject() coerced.  The
+    // coerced value is what's returned by the function.
+    var obj = Object.assign(812, { test: "something" });
+    print(obj instanceof Number);
```

Comment:
Right, it's just maybe not intuitive that `1 instanceof Number` is false but `new Number(1) instanceof Number` is true. At least it isn't for me :-)